### PR TITLE
drop 2.6 support

### PR DESCRIPTION
**Pull request by manishtomar**
Based on discussion in #74.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -12,7 +12,7 @@ the effects (that is, IO or state manipulation) in your code. Documentation is
 available at https://effect.readthedocs.org/, and its PyPI page is
 https://pypi.python.org/pypi/effect.
 
-It `supports`_ both Python 2.6 and up, and 3.4 and up, as well as PyPy.
+It `supports`_ both Python 2.7 and up, and 3.5 and up, as well as PyPy.
```
Comment (radix) [inline README.rst]: The change from 3.4 to 3.5 here is incorrect, since 3.4 is still supported.

Comment (manishtomar) [inline README.rst]: How about "supports both 2.7 and upto 3.5 as well as PyPy"?

Comment (radix) [inline README.rst]: it doesn't support 3.2 or 3.3 though.

Comment (radix) [inline README.rst]: (or 3.0 or 3.1...)

Comment (manishtomar) [inline README.rst]: Right. I am just going to make it explicit: "It `supports`_ Python 2.7, 3.4 and 3.5 as well as PyPy." :)

Comment (radix) [inline README.rst]: sounds good :)

**Pull request merged**